cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -1150,14 +1154,14 @@ std::vector<std::variant<ReadVersionOutput, DataError>> LocalVersionedEngine::te
 std::vector<std::variant<ReadVersionOutput, DataError>> LocalVersionedEngine::batch_read_internal(
     const std::vector<StreamId>& stream_ids,
     const std::vector<VersionQuery>& version_queries,
-    std::vector<ReadQuery>& read_queries,
+    std::vector<std::shared_ptr<ReadQuery>>& read_queries,
```

Comment:
What was the problem that required a shared_ptr to the ReadQuery? Did we have lifetime problems with it in a more async context?

Reply:
This function modifies the passed in read queries, and these modifications need to be visible in the Python layer for `_post_process_dataframe` to work properly when the non-query builder branch is taken